libcxx.mixin

```diff
@@ -0,0 +1,10 @@
+build:
+  libcxx:
+    cmake-args:
+      - "-DCMAKE_C_COMPILER=clang-6.0"
+      - "-DCMAKE_CXX_COMPILER=clang++-6.0"
+      - "-DCMAKE_CXX_FLAGS=-stdlib=libc++ -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS"
+      - "-DFORCE_BUILD_POCO=ON"
+      - "-DFORCE_BUILD_TINYXML=ON"
+      - "--no-warn-unused-cli"
+    merge-install: true
```

Comment:
What makes the merged install necessary when using libcxx?

Reply:
I was wrong, it is not, but my thinking I needed it was a symptom of a more generic problem I am having a hard time figuring out. 
* Build completely from clean
* `libPocoFoundation.so` is output to `install/lib`
* Once the build gets to `test_msgs`, the `LD_LIBRARY_PATH` does not seem to be set properly 
    * it links agains `install/lib/librosidl_typesupport_c.so`, which refers to `libPocoFoundation.so` 
    * `librosidl_typesupport_c.so` has no `RPATH` after being installed by CMake
    * `/usr/local/lib/libPocoFoundation.so` system package is chosen instead of the one under `install/`
* after this failed build with `std::`-related linker errors, if I source `install/setup.bash`, _then_ the next build succeeds, because setup sets `LD_LIBRARY_PATH`

So, how do I have the build set `LD_LIBRARY_PATH` properly so that the build can succeed on the first run through?

Reply:
This build works reliably the first time in the `ci` package's Docker container, but not on my "bare metal" ec2 instance - I'll investigate what's different about the configurations, it's not related to this review.